packages/sync-actions/test/product-sync-prices.spec.js

```diff
@@ -167,4 +167,551 @@ describe('Actions', () => {
+          channel: { typeId: 'channel', id: 'ch1' },
+        },
+        {
+          id: '222',
+          value: { currencyCode: 'USD', centAmount: 5000 },
+          country: 'US',
+          customerGroup: { typeId: 'customer-group', id: 'cg1' },
+        },
+        {
+          id: '333',
+          value: { currencyCode: 'SEK', centAmount: 10000 },
+          country: 'US',
+          channel: { typeId: 'channel', id: 'ch1' },
+        },
+        {
+          id: '666',
+          value: { currencyCode: 'GBP', centAmount: 1000 },
+          country: 'UK',
+          validFrom: dateNow,
+          validUntil: twoWeeksFromNow,
+        },
+      ]
+
+      now.masterVariant.prices = [
+        {
+          value: { currencyCode: 'EUR', centAmount: 4000 },
+          country: 'US',
+          customerGroup: { typeId: 'customer-group', id: 'cg1' },
+          channel: { typeId: 'channel', id: 'ch1' },
+        },
+        {
+          value: { currencyCode: 'USD', centAmount: 6000 },
+          country: 'US',
+          customerGroup: { typeId: 'customer-group', id: 'cg1' },
+        },
+        {
+          value: { currencyCode: 'SEK', centAmount: 15000 },
+          country: 'US',
+          channel: { typeId: 'channel', id: 'ch1' },
+        },
+        {
+          value: { currencyCode: 'GBP', centAmount: 10000 },
+          country: 'UK',
+          validFrom: dateNow,
+          validUntil: twoWeeksFromNow,
+        },
+      ]
+
+      const actions = productsSync.buildActions(now, before)
+
+      expect(actions).toEqual([
+        {
+          action: 'changePrice',
+          priceId: '111',
+          price: {
+            id: '111',
+            value: { currencyCode: 'EUR', centAmount: 4000 },
+            country: 'US',
+            customerGroup: { typeId: 'customer-group', id: 'cg1' },
+            channel: { typeId: 'channel', id: 'ch1' },
+          },
+        },
+        {
+          action: 'changePrice',
+          priceId: '222',
+          price: {
+            id: '222',
+            value: { currencyCode: 'USD', centAmount: 6000 },
+            country: 'US',
+            customerGroup: { typeId: 'customer-group', id: 'cg1' },
+          },
+        },
+        {
+          action: 'changePrice',
+          priceId: '333',
+          price: {
+            id: '333',
+            value: { currencyCode: 'SEK', centAmount: 15000 },
+            country: 'US',
+            channel: { typeId: 'channel', id: 'ch1' },
+          },
+        },
+        {
+          action: 'changePrice',
+          priceId: '666',
+          price: {
+            id: '666',
+            value: { currencyCode: 'GBP', centAmount: 10000 },
+            country: 'UK',
+            validFrom: dateNow,
+            validUntil: twoWeeksFromNow,
+          },
+        },
+      ])
+    })
+
+    it('should build changePrice actions to update dates when validFrom or validUntil priceSelection matches and there is no priceId', () => {
```

Comment:
The "when" part is a `describe`.

So 

```txt
describe: when `validForm` or `validFrom` match
  describe: without `priceId`
     it: should build `changePrice` actions
```

This way we can also build a way nicer "tree" of expectations which reads super nice when running the tests.